<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>770</width>
    <height>800</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>769</width>
    <height>340</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>770</width>
    <height>800</height>
   </size>
  </property>
  <property name="cursor">
   <cursorShape>PointingHandCursor</cursorShape>
  </property>
  <property name="windowTitle">
   <string>Robot</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="minimumSize">
    <size>
     <width>275</width>
     <height>275</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>276</width>
     <height>276</height>
    </size>
   </property>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>770</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
   </widget>
   <widget class="QMenu" name="menuProgram">
    <property name="title">
     <string>Program</string>
    </property>
    <addaction name="actionAdd_Step"/>
    <addaction name="actionEnd_Step"/>
    <addaction name="actionRun_Program"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuProgram"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="iconSize">
    <size>
     <width>16</width>
     <height>16</height>
    </size>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNew"/>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
   <addaction name="separator"/>
   <addaction name="actionAdd_Step"/>
   <addaction name="actionEnd_Step"/>
   <addaction name="separator"/>
   <addaction name="actionRun_Program"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QDockWidget" name="dwServoControl">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="minimumSize">
    <size>
     <width>250</width>
     <height>274</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>251</width>
     <height>275</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>Servo Control</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <widget class="QSlider" name="servo_1">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>160</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="value">
      <number>128</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QSlider" name="servo_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>160</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="value">
      <number>128</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QSlider" name="servo_6">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>210</y>
       <width>160</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="value">
      <number>128</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QSlider" name="servo_3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>160</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="value">
      <number>128</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QSlider" name="servo_4">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>130</y>
       <width>160</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="value">
      <number>128</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QSlider" name="servo_5">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>170</y>
       <width>160</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="value">
      <number>128</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="lblServo_1">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>13</y>
       <width>31</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>128</string>
     </property>
    </widget>
    <widget class="QLabel" name="lblServo_2">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>53</y>
       <width>31</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>128</string>
     </property>
    </widget>
    <widget class="QLabel" name="lblServo_3">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>93</y>
       <width>31</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>128</string>
     </property>
    </widget>
    <widget class="QLabel" name="lblServo_4">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>133</y>
       <width>31</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>128</string>
     </property>
    </widget>
    <widget class="QLabel" name="lblServo_5">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>173</y>
       <width>31</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>128</string>
     </property>
    </widget>
    <widget class="QLabel" name="lblServo_6">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>213</y>
       <width>31</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>128</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidget_3">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="minimumSize">
    <size>
     <width>180</width>
     <height>274</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>181</width>
     <height>275</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>Program Step</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3">
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="lblProgramStep">
       <property name="text">
        <string>0</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QTextBrowser" name="textBrowser"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dwControlPanel">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="maximumSize">
    <size>
     <width>251</width>
     <height>524287</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>Control Panel</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <widget class="QSpinBox" name="spbX">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>81</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-36</number>
     </property>
     <property name="maximum">
      <number>36</number>
     </property>
     <property name="value">
      <number>20</number>
     </property>
    </widget>
    <widget class="QSpinBox" name="spbY">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>81</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-36</number>
     </property>
     <property name="maximum">
      <number>36</number>
     </property>
     <property name="value">
      <number>0</number>
     </property>
    </widget>
    <widget class="QSpinBox" name="spbZ">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>81</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-36</number>
     </property>
     <property name="maximum">
      <number>36</number>
     </property>
     <property name="value">
      <number>10</number>
     </property>
    </widget>
    <widget class="QSpinBox" name="spbP">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>100</y>
       <width>81</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-180</number>
     </property>
     <property name="maximum">
      <number>180</number>
     </property>
    </widget>
    <widget class="QLabel" name="lblX">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>15</y>
       <width>47</width>
       <height>10</height>
      </rect>
     </property>
     <property name="text">
      <string>X</string>
     </property>
    </widget>
    <widget class="QLabel" name="lblY">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>44</y>
       <width>47</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>Y</string>
     </property>
    </widget>
    <widget class="QLabel" name="lblZ">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>74</y>
       <width>47</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>Z</string>
     </property>
    </widget>
    <widget class="QLabel" name="lblP">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>103</y>
       <width>47</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>P</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>390</y>
       <width>221</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>TextLabel</string>
     </property>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>360</y>
       <width>231</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>TextLabel</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>140</y>
       <width>231</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>TextLabel</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>170</y>
       <width>231</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>TextLabel</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>200</y>
       <width>231</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>TextLabel</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>230</y>
       <width>231</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>TextLabel</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dwMode">
   <property name="maximumSize">
    <size>
     <width>150</width>
     <height>524287</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>Mode</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_6">
    <widget class="QComboBox" name="cbbMode">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>101</width>
       <height>22</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>Servo Control</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Control Panel</string>
      </property>
     </item>
    </widget>
    <widget class="QPushButton" name="btnStart">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>60</y>
       <width>75</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Start</string>
     </property>
    </widget>
    <widget class="QComboBox" name="cbbSerialPort">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>160</y>
       <width>101</width>
       <height>22</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>Wybierz port</string>
      </property>
     </item>
    </widget>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidget">
   <property name="windowTitle">
    <string>Plot</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_4">
    <widget class="QCustomPlot" name="plotFront" native="true">
     <property name="geometry">
      <rect>
       <x>9</x>
       <y>9</y>
       <width>341</width>
       <height>221</height>
      </rect>
     </property>
    </widget>
    <widget class="QCustomPlot" name="plotAbove" native="true">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>240</y>
       <width>341</width>
       <height>191</height>
      </rect>
     </property>
    </widget>
   </widget>
  </widget>
  <action name="actionNew">
   <property name="icon">
    <iconset resource="icons/icons.qrc">
     <normaloff>:/new.png</normaloff>:/new.png</iconset>
   </property>
   <property name="text">
    <string>New</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="icons/icons.qrc">
     <normaloff>:/open.png</normaloff>:/open.png</iconset>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionAdd_Step">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="icons/icons.qrc">
     <normaloff>:/add.png</normaloff>:/add.png</iconset>
   </property>
   <property name="text">
    <string>Add Step</string>
   </property>
  </action>
  <action name="actionEnd_Step">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="icons/icons.qrc">
     <normaloff>:/stop.png</normaloff>:/stop.png</iconset>
   </property>
   <property name="text">
    <string>End Step</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="icons/icons.qrc">
     <normaloff>:/save.png</normaloff>:/save.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionRun_Program">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="icons/icons.qrc">
     <normaloff>:/run.png</normaloff>:/run.png</iconset>
   </property>
   <property name="text">
    <string>Run Program</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header location="global">qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="icons/icons.qrc"/>
 </resources>
 <connections/>
</ui>
